docs/remote/devcontainerjson-reference.md

```diff
@@ -59,6 +59,12 @@ While some devcontainer.json properties apply generally, others are only used in
 
 If you've already built the container and connected to it, be sure to run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change.
 
+> **Note:** The format of certain properties will vary depending on the involvement of a shell.
+
+> `postCreateCommand`, `postStartCommand`, `postAttachCommand`, and `initializeCommand` all have an array and a string type. An array is passed to the OS for execution without going through a shell, whereas a string goes through a shell (it needs to be parsed into command and arguments).
```

Comment:
Maybe mention "runArgs" here too to make this paragraph complete as a high-level description. (Maybe in separate sentence since "runArgs" only has the array variant.)

Reply:
That's good feedback, so that it hopefully won't feel as though `runArgs` is randomly introduced later. I updated the section here, let me know if you think it helps, or I can tweak it a bit more.